Update link location, notification method, and namespace check

Eejit43 opened the pull request:
This PR makes multiple changes:

* Updates the method of adding the start link from appending to the left sidebar, to added to the "More"/"Tools" dropdown menu (using `mw.util.addPortletLink`)
* Improves the notification method, using `mw.notify` instead of a custom display
* Only runs script in non-virtual (Special, Media) namespaces (replaces commented-out check for just mainspace, draftspace, and sandboxes) (closes #117)

NovemLinguae commented:
Thanks for the PR

1) Running this code on https://en.wikipedia.org/w/index.php?title=Draft:Greenwitchcom_(aka_Isabella_Ferrari)&oldid=1140345466 gets the following error

<img width="464" alt="2023-02-19_095425" src="https://user-images.githubusercontent.com/79697282/219965965-f804511a-ee32-4e6b-b918-ae8ccb3e6e0a.png">

2) I like the link being at the bottom of the top left menu (p-navigation). Can we place it there instead of in p-tb?

2a) If you want, you can add code that checks `if ( window.draftCleanerPlaceLinkInToolsMenu )`, and if that is set, place it in the tools menu instead. That would let users such as yourself configure it by placing `window.draftCleanerPlaceLinkInToolsMenu = true;` in their common.js.

NovemLinguae commented:
3) `namespaceNumber` is not defined

<img width="563" alt="image" src="https://user-images.githubusercontent.com/79697282/219966282-d1285b31-8b5c-4037-b0f4-46fa246e9ab9.png">

4) `showClickableButton()` is not defined

<img width="274" alt="image" src="https://user-images.githubusercontent.com/79697282/219966265-bac023cf-a04c-42d3-8176-dd71d7fa0c63.png">

NovemLinguae commented:
5) You copy pasted some function code with `return` outside a function, so the returns won't work anymore.

<img width="621" alt="image" src="https://user-images.githubusercontent.com/79697282/219966721-5ac78df2-042c-4825-a71e-6cb0343c89bf.png">

Eejit43 commented:
1) Fixed
2) Done, however using `p-cactions` instead of `p-tb`, which is generally the standard point for such additions
3) Fixed
4) Fixed
5) Will fix

NovemLinguae commented:
Thanks! By the way, never mind about 5. I forgot that when I run my deploy script, it wraps everything in `$(async function() {`, so those returns are in a function after all.

Eejit43 commented:
Yep was about to say that, thanks! 👍

NovemLinguae commented:
Looks like we have some users that depend on ` window.draftCleanerPutInToolsMenu = true;` to put the link in the tools menu (confirmed with [wiki insource search](https://en.wikipedia.org/w/index.php?search=insource%3A%22window.draftCleanerPutInToolsMenu%22&title=Special%3ASearch&profile=advanced&fulltext=1&ns2=1)), so I've added ` window.draftCleanerPutInMoreMenu = true;` for your use case. Go ahead and change your common.js if you'd like.

Anyway this is ready to merge. Thanks for the PR. I appreciate you splitting the workload :)
